packages/sdk/src/internal/messaging/data-service-messaging.ts

```diff
@@ -111,21 +111,43 @@ export class DataServiceMessaging implements Messaging {
     const otherMembersPks = Object.fromEntries(
       otherThreadMembers.map((member) => [member.publicKey.toBase58(), member]),
     );
+    
+    let thisThreadMember = {
+      publicKey: new PublicKey(meMember.publicKey),
+      scopes: fromDataServiceScopes(meMember.scopes),
+      // lastReadMessageTimestamp: new Date(), // TODO: implement
+    };
+    let messages = dialect.messages.map((message) => (
+      message
+    ));
+    let lastMessage = messages[messages.length - 1] || null;
```

Comment:
thinking something like this
`    let lastMessage = messages.length > 0 ? messages[messages.length - 1] : null;
    let lastThreadMessage: ThreadMessage | null = null;
    if (lastMessage != null) {
      lastThreadMessage = {
        text: serde.deserialize(new Uint8Array(lastMessage.text)),
        timestamp: new Date(lastMessage.timestamp),
        author: lastMessage.owner === this.me.toBase58()
          ? thisThreadMember
          : otherMembersPks[lastMessage.owner]!,
        deduplicationId: lastMessage.deduplicationId ,
      }
    };`

Reply:
also have you verified [messages.length - 1] is the latest message? (vs. messages[0])

Reply:
> also have you verified [messages.length - 1] is the latest message? (vs. messages[0])

I haven't. I still need to figure that out